src/types.ts

```diff
@@ -739,6 +739,14 @@ export class ZodString extends ZodType<string, ZodStringDef> {
       ...errorUtil.errToObj(options?.message),
     });
   }
+  
+  numericString(message?: errorUtil.ErrMessage) {
+    return this._addCheck({
+      kind: "regex",
```

Comment:
Should align with other regex QoL methods and add its own kind `"numeric"` and `isNumeric()` method.